plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/server/wst/LiveReloadServerBehaviour.java

```diff
@@ -325,25 +325,26 @@ private void startProxy(final IServer startedServer) {
 		}
 		// create a new proxy
 		else {
-			final int targetPort = WSTUtils.getWebPort(startedServer);
-			if (targetPort != -1) {
-				try {
-					// now, let's init and start the embedded jetty proxy server
-					// from the
-					// server attributes
-					final boolean allowRemoteConnections = isRemoteConnectionsAllowed();
-					final boolean enableScriptInjection = isScriptInjectionEnabled();
-					final String proxyHost = getProxyHost();
-					final int proxyPort = getProxyPort(startedServer);
-					final LiveReloadProxyServer proxyServer = new LiveReloadProxyServer(proxyHost, proxyPort, startedServer.getHost(), targetPort,
-							websocketPort, allowRemoteConnections, enableScriptInjection);
-					proxyServers.put(startedServer, proxyServer);
-					final JettyServerRunner proxyRunner = JettyServerRunner.start(proxyServer);
-					proxyRunners.put(startedServer, proxyRunner);
-				} catch (Exception e) {
-					Logger.error("Failed to create or start LiveReload proxy for server " + startedServer.getName(), e);
+			try {
+				final int targetPort = WSTUtils.getWebPort(startedServer);
+				if (targetPort == -1) {
+					return;
 				}
-
+				final String targetHost = WSTUtils.getWebHost(startedServer);
+				// now, let's init and start the embedded jetty proxy server
+				// from the
+				// server attributes
+				final boolean allowRemoteConnections = isRemoteConnectionsAllowed();
+				final boolean enableScriptInjection = isScriptInjectionEnabled();
+				final String proxyHost = getProxyHost();
+				final int proxyPort = getProxyPort(startedServer);
+				final LiveReloadProxyServer proxyServer = new LiveReloadProxyServer(proxyHost, proxyPort, targetHost, targetPort,
+						websocketPort, allowRemoteConnections, enableScriptInjection);
+				proxyServers.put(startedServer, proxyServer);
+				final JettyServerRunner proxyRunner = JettyServerRunner.start(proxyServer);
+				proxyRunners.put(startedServer, proxyRunner);
+			} catch (Exception e) {
+				Logger.error("Failed to create or start LiveReload proxy for server " + startedServer.getName(), e);

```

Comment:
we can't since this is a 'core' module that has no access to UI features. That would require more significant changes/refactoring, I believe